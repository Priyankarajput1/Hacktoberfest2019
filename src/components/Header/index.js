import PropTypes from 'prop-types';
import React from 'react';
import styled from '@emotion/styled';

import githubCorner from '../../images/github_corner.svg';

const GithubButton = styled.a`
  height: 100%;
  display: block;
  position: absolute;
  right: 0;
  top: 0;
  & > img {
    height: 100%;
  }
  transition: all 0.4s cubic-bezier(0.165, 0.84, 0.44, 1);
  &:hover {
    transform: translate(20px, -20px);
  }
`;

const Wrapper = styled.header`
  grid-area: header;
  position: fixed;
  top: 0;
  width: 100%;
  left: 0;
  right: 0;
  height: 200px;
`;

const Header = () => (
  <Wrapper>
    <GithubButton
      target="_blank"
      href="https://github.com/greenberrynl/hacktoberfest2019"
    >
      <img src={githubCorner} alt="Github" />
    </GithubButton>
  </Wrapper>
);

Header.propTypes = {};

Header.defaultProps = {};

export default Header;
