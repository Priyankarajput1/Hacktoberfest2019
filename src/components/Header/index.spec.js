import React from 'react';
import renderer from 'react-test-renderer';

import Header from './index';

describe('components/Header', () => {
  it('should render', () => {
    const component = renderer.create(<Header>Hello Header</Header>);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
