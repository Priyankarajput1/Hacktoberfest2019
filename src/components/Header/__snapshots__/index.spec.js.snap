// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header should render 1`] = `
.emotion-2 {
  grid-area: header;
  position: fixed;
  top: 0;
  width: 100%;
  left: 0;
  right: 0;
  height: 200px;
}

.emotion-0 {
  height: 100%;
  display: block;
  position: absolute;
  right: 0;
  top: 0;
  -webkit-transition: all 0.4s cubic-bezier(0.165,0.84,0.44,1);
  transition: all 0.4s cubic-bezier(0.165,0.84,0.44,1);
}

.emotion-0 > img {
  height: 100%;
}

.emotion-0:hover {
  -webkit-transform: translate(20px,-20px);
  -ms-transform: translate(20px,-20px);
  transform: translate(20px,-20px);
}

<header
  className="emotion-2 emotion-3"
>
  <a
    className="emotion-0 emotion-1"
    href="https://github.com/greenberrynl/hacktoberfest2019"
    target="_blank"
  >
    <img
      alt="Github"
      src="totally real file"
    />
  </a>
</header>
`;
